data/src/ctcp.rs

```diff
@@ -0,0 +1,26 @@
+#[derive(Debug)]
+pub struct Query<'a> {
+    pub command: &'a str,
+    pub params: &'a str,
+}
+
+pub fn is_ctcp_query(text: &str) -> bool {
+    text.starts_with('\u{1}')
+}
+
+pub fn parse_ctcp_query(text: &str) -> Result<Query, &'static str> {
```

Comment:
Nit: I don't think Result gives us anything, plus we always .ok it. Let's make this an Option